File: app.client/src/app/user-application/tasks/model/task.ts
import { User } from "../../user/model/user"

export interface Task {
    id: string,
    title: string,
    description?: string,
    createdAt?: Date,
    updatedAt?: Date,
    status: TaskStatus,
    priority?: TaskPriority,
    deadline?: Date,
    createdBy?: User;
}

export enum TaskPriority {
    NONE = 'None',
    LOW = 'Low',
    MEDIUM = 'Medium',
    HIGH = 'High'
}

export enum TaskStatus {
    NEW = 'New',
    IN_PROGRESS = "InProgress",
    DONE = 'Done'
}

export interface TaskCreatePayload {
    title: string,
    description?: string,
    status?: TaskStatus,
    priority?: TaskPriority,
    deadline?: Date,
    createdBy?: User;
}

export interface TaskUpdatePayload {
    title?: string,
    description?: string,
    status?: TaskStatus,
    priority?: TaskPriority,
    deadline?: Date
}

File: App.Server/Model/Task.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;

namespace App.Server.Model
{
    public class Task
    {
        public string Id { get; set; }

        public string Title { get; set; }

        public string? Description { get; set; }

        [Column(TypeName = "varchar(255)")]
        public TaskStatus Status { get; set; }

        [Column(TypeName = "varchar(255)")]
        [AllowNull]
        public TaskPriority Priority { get; set; }

        [DataType(DataType.DateTime)]
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public DateTime? Deadline { get; set; }

    }

    public enum TaskStatus
    {
        New,
        InProgress,
        Done
    }

    public enum TaskPriority
    {
        None,
        Low,
        Medium,
        High
    }
}
